src/App.tsx

```diff
@@ -1,26 +1,99 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import { deleteTodos, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { FilterOptions } from './FilterOptions';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      try {
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } catch (error) {
+        setErrorMessage('Unable to load todos');
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timer = setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [errorMessage]);
+
+  const handleTodoDelete = (todoId: number) => {
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      });
+  };
+
+  const handleCompletedTodosDeleted = () => {
+    setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+  };
+
+  const handleTodosToggle = () => {
+    const hasIncompleteTodos = todos.some(todo => !todo.completed);
+
+    setTodos(currentTodos =>
+      currentTodos.map(todo => ({
+        ...todo,
+        completed: hasIncompleteTodos,
+      })),
+    );
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case FilterOptions.ACTIVE:
+        return !todo.completed;
+      case FilterOptions.COMPLETED:
+        return todo.completed;
+      default:
+        return true;
```

Comment:
Your solution means that you are going to use method 'filter' also in case of state filter === all.  It leads you to overiterating and  missing the optimisation. Imagine that your array will have too many items inside and it will take more time to render the page. So you can create a function getFilteredTodos and use your switch case and filter method in it with a default value of 'return todos'   . After that const filteredTodos = getFilteredTodos()